osu.Game.Tests/Visual/Multiplayer/TestSceneMultiSpectatorScreen.cs

```diff
@@ -0,0 +1,387 @@
+        private BeatmapInfo importedBeatmap;
+        private int importedBeatmapId;
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            importedSet = ImportBeatmapTest.LoadOszIntoOsu(game, virtualTrack: true).Result;
+            importedBeatmap = importedSet.Beatmaps.First(b => b.RulesetID == 0);
+            importedBeatmapId = importedBeatmap.OnlineBeatmapID ?? -1;
+        }
+
+        public override void SetUpSteps()
+        {
+            base.SetUpSteps();
+
+            AddStep("reset sent frames", () => nextFrame.Clear());
+
+            AddStep("add streaming client", () =>
+            {
+                Remove(streamingClient);
+                Add(streamingClient);
+            });
+
+            AddStep("finish previous gameplay", () =>
+            {
+                foreach (var id in playingUserIds)
+                    streamingClient.EndPlay(id, importedBeatmapId);
+                playingUserIds.Clear();
+            });
+        }
+
+        [Test]
+        public void TestDelayedStart()
+        {
+            AddStep("start players silently", () =>
+            {
+                Client.CurrentMatchPlayingUserIds.Add(55);
+                Client.CurrentMatchPlayingUserIds.Add(56);
+                playingUserIds.Add(55);
+                playingUserIds.Add(56);
+                nextFrame[55] = 0;
+                nextFrame[56] = 0;
+            });
+
+            loadSpectateScreen(false);
+
+            AddWaitStep("wait a bit", 10);
+            AddStep("load player 55", () => streamingClient.StartPlay(55, importedBeatmapId));
+            AddUntilStep("one player added", () => spectatorScreen.ChildrenOfType<Player>().Count() == 1);
+
+            AddWaitStep("wait a bit", 10);
+            AddStep("load player 56", () => streamingClient.StartPlay(56, importedBeatmapId));
+            AddUntilStep("two players added", () => spectatorScreen.ChildrenOfType<Player>().Count() == 2);
+        }
+
+        [Test]
+        public void TestGeneral()
+        {
+            int[] userIds = Enumerable.Range(0, 4).Select(i => 55 + i).ToArray();
+
+            start(userIds);
+            loadSpectateScreen();
+
+            sendFrames(userIds, 1000);
+            AddWaitStep("wait a bit", 20);
+        }
+
+        [Test]
+        public void TestPlayersMustStartSimultaneously()
+        {
+            start(new[] { 55, 56 });
+            loadSpectateScreen();
+
+            // Send frames for one player only, both should remain paused.
+            sendFrames(55, 20);
+            checkPausedInstant(55, true);
+            checkPausedInstant(56, true);
+
+            // Send frames for the other player, both should now start playing.
+            sendFrames(56, 20);
+            checkPausedInstant(55, false);
+            checkPausedInstant(56, false);
+        }
+
+        [Test]
+        public void TestPlayersDoNotStartSimultaneouslyIfBufferingForMaximumStartDelay()
```

Comment:
This test fails if run in isolation (have tested this was the case before my renaming changes):

![20210426 164637 (JetBrains Rider)](https://user-images.githubusercontent.com/191335/116047377-fa5f4500-a6ae-11eb-8e16-f13038177415.png)